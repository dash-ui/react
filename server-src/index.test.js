// import {render as renderComponent} from '@testing-library/react'
import defaultStyles from '@-ui/styles'
import {toComponent} from './index'

afterEach(() => {
  document.getElementsByTagName('html')[0].innerHTML = ''
})

describe('<Style>', () => {
  const html = style => `
    <div class=${style('flex')}>
      <div class=${style('btn')}>
        Hello
      </div>
    </div>
  `

  it.skip('renders a style tag', () => {
    const myStyles = defaultStyles.create({})
    const style = myStyles({
      flex: {display: 'flex'},
      btn: `
        border-radius: 1000px;
        background: blue;
        color: white;
      `,
    })

    style('flex')
    style('btn')

    expect(toComponent(html(style), myStyles)).toMatchSnapshot()
  })
})
