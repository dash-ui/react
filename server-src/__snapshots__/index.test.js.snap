// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Style> renders a style tag 1`] = `
<Style
  html="
    <div class=dash-1ut9bc3>
      <div class=dash-1i986zm>
        Hello
      </div>
    </div>
  "
  styles={[Function]}
/>
`;
